index.js

```diff
@@ -31,23 +32,39 @@ exports.handler = async event => {
     // kinesis downloads/impressions to be logged in batch
     const kinesisRecords = []
 
+    // only assumerole once for dynamodb access
+    const ddbClient = await dynamo.client()
+
+    // concurrently lookup arrangement json
+    const digests = decoded.map(d => d.digest).reduce((acc, d) => ({ ...acc, [d]: null }), {})
+    const loadArrangements = Object.keys(digests).map(async digest => {
+      try {
+        digests[digest] = await Arrangement.load(digest, redis, ddbClient)
```

Comment:
Does it need to `await` here if it's also awaiting on the `Promise.all`?

Reply:
Yes, because I want to set `digests[digest]` to the actual result, not the promise.  (Also, the surrounding try/catch block wouldn't do anything without an await here).